fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/rules/AbstractDataAccessRule.java

```diff
@@ -256,7 +256,6 @@ private static List<PartialMatchWithCompensation> maximumCoverageMatches(@Nonnul
         final var partialMatchesWithCompensation =
                 matches
                         .stream()
-                        .filter(partialMatch -> partialMatch.getMatchCandidate() instanceof WithPrimaryKeyMatchCandidate)
```

Comment:
So this will allow you to use the same access rule for aggregate indexes. Got it.